model/host/host.go

```diff
@@ -544,24 +550,40 @@ func (h *Host) SetProvisioning() error {
 	)
 }
 
-func (h *Host) SetDecommissioned(user string, logs string) error {
+// SetDecommissioned sets the host as decommissioned. If checkTaskGroup is set,
+// we only update the host if it hasn't started running a task group.
+func (h *Host) SetDecommissioned(user string, checkTaskGroup bool, logs string) error {
+	query := bson.M{}
+	if checkTaskGroup {
+		query[RunningTaskGroupKey] = bson.M{"$eq": ""}
+	}
 	if h.HasContainers {
 		containers, err := h.GetContainers()
 		grip.Error(message.WrapError(err, message.Fields{
 			"message": "error getting containers",
 			"host_id": h.Id,
 		}))
+		catcher := grip.NewBasicCatcher()
+		failedContainerIds := []string{}
 		for _, c := range containers {
-			err = c.SetStatus(evergreen.HostDecommissioned, user, "parent is being decommissioned")
+			err = c.setStatusAndFields(evergreen.HostDecommissioned, query, nil, user, "parent is being decommissioned")
 			if err != nil && err.Error() != ErrorHostAlreadyTerminated {
-				grip.Warning(message.WrapError(err, message.Fields{
-					"message": "error decommissioning container",
-					"host_id": c.Id,
-				}))
+				catcher.Add(err)
+				failedContainerIds = append(failedContainerIds, c.Id)
 			}
 		}
+		grip.Warning(message.WrapError(catcher.Resolve(), message.Fields{
+			"message":  "error decommissioning containers",
+			"host_ids": failedContainerIds,
+		}))
 	}
-	return h.SetStatus(evergreen.HostDecommissioned, user, logs)
+	err := h.setStatusAndFields(evergreen.HostDecommissioned, query, nil, user, logs)
+	// Shouldn't consider it an error if the host isn't found when checking task group,
+	// because a task group may have been set for the host.
+	if err != nil && checkTaskGroup && !adb.ResultsNotFound(err) {
```

Comment:
Is it supposed to return the error if `checkTaskGroup` is false?

Reply:
ah yes my bad